File: algo_cpp/bj1214.cpp
#include<cstdio>
#include<algorithm>
using namespace std;


int main(void) {
	int d, p, q;
	scanf("%d %d %d", &d, &p, &q);
	if (d % p == 0 || d % q == 0) {
		printf("%d", d);
		return 0;
	}
	if (p < q) {
		int t = p;
		p = q;
		q = t;
	}
	int r = 2e9;
	for (int i = 0; i < min(q,d / p + 2); i++) {
		int tq = (d - p * i) / q;
		for (int j = max(0, tq - 2); j < tq + 2; j++) {
			int tr = p * i + q * j;
			if (tr < d)
				continue;
			r = min(r, tr);
		}
	}
	printf("%d", r);
	return 0;
}

File: algo_py/boj/bj1214.py
d, p, q = map(int, input().split())
if 0 in (d % p, d % q, d % (p+q)):
    print(d)
    exit()
if p < q:
    p, q = q, p
r = 2e9
for i in range(min(q, d//p+2)):
    tq = (d-p*i)//q
    for j in range(max(0, tq-2), tq+2):
        if (tr := p*i+q*j) < d:
            continue
        r = min(r, tr)
print(r)
